src/pocketmine/entity/Entity.php

```diff
@@ -898,6 +899,13 @@ public function attack(EntityDamageEvent $source) : void{
 		$this->setHealth($this->getHealth() - $source->getFinalDamage());
 	}
 
+	/**
+	 * @param PlayerGentlyTouchEntityEvent $source
+	 */
+	public function interact(PlayerGentlyTouchEntityEvent $source) : void{
```

Comment:
Please let's not have another godawful mess like `Entity->attack()`. The intent of this function is not clear from its name. Is the entity interacting with some other entity, or is it the one getting interacted with?